Cannot build from source [help wanted]

$ git clone https://code.google.com/p/closure-stylesheets/
$ cd closure-stylesheets
$ mvn compile assembly:single

[INFO] Scanning for projects...

*****

[INFO] 
[INFO] --- maven-assembly-plugin:2.3:single (default-cli) @ closure-stylesheets ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 10.066s
[INFO] Finished at: Tue Aug 30 15:26:41 MSK 2016
[INFO] Final Memory: 16M/224M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.3:single (default-cli) on project closure-stylesheets: Error reading assemblies: No assembly descriptors found. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

@iflan 

Updating maven does not help

$ sudo apt-get purge maven maven2 maven3
$ sudo apt-add-repository ppa:andrei-pozolotin/maven3
$ sudo apt-get update
$ sudo apt-get install maven3
$ mvn -v

Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T19:41:47+03:00)
Maven home: /usr/share/maven3
Java version: 1.7.0_111, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.19.0-51-generic", arch: "amd64", family: "unix"

$ mvn compile assembly:single

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 15.598 s
[INFO] Finished at: 2016-08-30T16:28:50+03:00
[INFO] Final Memory: 24M/352M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.2-beta-5:single (default-cli) on project closure-stylesheets: Error reading assemblies: No assembly descriptors found. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

--- a/README.md
+++ b/README.md
@@ -21,7 +21,7 @@ java -jar closure-stylesheets.jar --help
 ```
 
 [download]: https://github.com/google/closure-stylesheets/releases
-[build it from source]: https://github.com/google/closure-stylesheets/wiki/Building From Source.md
+[build it from source]: https://github.com/google/closure-stylesheets/wiki/Building-From-Source.md
 
 
 ## CSS Extensions

--- a/pom.xml
+++ b/pom.xml
@@ -119,7 +119,6 @@
             <configuration>
               <jdkVersion>${java.version}</jdkVersion>
               <javadocFriendlyComments>true</javadocFriendlyComments>
-              <packageName>org.javacc.parser</packageName>
               <sourceDirectory>${project.basedir}/src</sourceDirectory>
               <outputDirectory>${project.build.directory}/generated-sources</outputDirectory>
             </configuration>

--- a/pom.xml
+++ b/pom.xml
@@ -148,9 +148,7 @@
     <dependency>
       <groupId>args4j</groupId>
       <artifactId>args4j</artifactId>
-      <!-- 2.0.17 is actually used,
-           but this is the closest thing in Maven Central Repo. -->
-      <version>2.0.16</version>
+      <version>2.0.26</version>
     </dependency>
 
     <dependency>
@@ -186,14 +184,14 @@
 
     <dependency>  <!-- For com.google.debugging.sourcemap -->
       <groupId>com.google.javascript</groupId>
-      <artifactId>closure-compiler</artifactId>
-      <version>v20160619</version>
+      <artifactId>closure-compiler-unshaded</artifactId>
+      <version>v20160713</version>
     </dependency>
 
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
-      <version>4.8.2</version>
+      <version>4.11</version>
       <scope>test</scope>
     </dependency>
 

--- a/pom.xml
+++ b/pom.xml
@@ -126,6 +126,21 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-assembly-plugin</artifactId>
+        <version>2.6</version>
+        <configuration>
+          <archive>
+            <manifest>
+              <mainClass>com.google.common.css.compiler.commandline.ClosureCommandLineCompiler</mainClass>
+            </manifest>
+          </archive>
+          <descriptorRefs>
+            <descriptorRef>jar-with-dependencies</descriptorRef>
+          </descriptorRefs>
+        </configuration>
+      </plugin>
     </plugins>
   </build>
 

--- a/src/com/google/common/css/compiler/ast/CssComponentNode.java
+++ b/src/com/google/common/css/compiler/ast/CssComponentNode.java
@@ -77,7 +77,7 @@ public class CssComponentNode extends CssAtRuleNode {
     return getType() == CssAtRuleNode.Type.ABSTRACT_COMPONENT;
   }
 
-  @SuppressWarnings("StringEquality")
+  @SuppressWarnings("ReferenceEquality")
   public boolean isImplicitlyNamed() {
     // Test against the exact string instance.
     return getName().getValue() == IMPLICIT_NODE_NAME;

--- a/src/com/google/common/css/compiler/ast/CssRulesetNode.java
+++ b/src/com/google/common/css/compiler/ast/CssRulesetNode.java
@@ -81,6 +81,7 @@ public class CssRulesetNode extends CssNode {
    */
   public CssRulesetNode(CssRulesetNode node) {
     this(node.getDeclarations().deepCopy());
+    this.setSourceCodeLocation(node.getSourceCodeLocation());
     this.setComments(node.getComments());
     this.selectors = node.getSelectors().deepCopy();
   }

--- a/src/com/google/common/css/compiler/ast/CssTreeBuilder.java
+++ b/src/com/google/common/css/compiler/ast/CssTreeBuilder.java
@@ -367,6 +367,7 @@ public class CssTreeBuilder implements
 
     Preconditions.checkState(ruleset == null);
     ruleset = new CssRulesetNode(comments);
+    ruleset.setSourceCodeLocation(selectorList.getSourceCodeLocation());
     ruleset.setSelectors(selectorList);
     comments.clear();
 

--- a/src/com/google/common/css/compiler/passes/BrowserPrefixGenerator.java
+++ b/src/com/google/common/css/compiler/passes/BrowserPrefixGenerator.java
@@ -289,6 +289,14 @@ final class BrowserPrefixGenerator {
         .addExpandPropertyValue("repeating-linear-gradient")
         .build());
 
+    builder.add(new BrowserPrefixRule.Builder()
+        .matchPropertyName("cursor")
+        .matchPropertyValue("grab")
+        .isFunction(false)
+        .addExpandPropertyValue("-moz-grab")
+        .addExpandPropertyValue("-webkit-grab")
+        .build());
+
     // Needed for Firefox 15, Chrome 25, Safari 6, iOS Safari 6.1 or less
     builder.add(new BrowserPrefixRule.Builder()
         .matchPropertyValue("calc")

--- a/src/com/google/common/css/compiler/passes/CheckMissingRequire.java
+++ b/src/com/google/common/css/compiler/passes/CheckMissingRequire.java
@@ -54,10 +54,10 @@ public final class CheckMissingRequire extends DefaultTreeVisitor implements Css
   private static final Logger logger = Logger.getLogger(CheckMissingRequire.class.getName());
 
   private static final Pattern OVERRIDE_SELECTOR_REGEX = Pattern.compile(
-      "/\\*\\s+@overrideSelector\\s+\\{(.*)\\}\\s+\\*/");
+      "/\\*\\*?\\s+@overrideSelector\\s+\\{(.*)\\}\\s+\\*/");
 
   private static final Pattern OVERRIDE_DEF_REGEX = Pattern.compile(
-      "/\\*\\s+@overrideDef\\s+\\{(.*)\\}\\s+\\*/");
+      "/\\*\\*?\\s+@overrideDef\\s+\\{(.*)\\}\\s+\\*/");
 
   private final VisitController visitController;
   private final ErrorManager errorManager;

--- a/src/com/google/common/css/compiler/passes/CheckMissingRequire.java
+++ b/src/com/google/common/css/compiler/passes/CheckMissingRequire.java
@@ -54,10 +54,10 @@ public final class CheckMissingRequire extends DefaultTreeVisitor implements Css
   private static final Logger logger = Logger.getLogger(CheckMissingRequire.class.getName());
 
   private static final Pattern OVERRIDE_SELECTOR_REGEX = Pattern.compile(
-      "/\\*\\*?\\s+@overrideSelector\\s+\\{(.*)\\}\\s+\\*/");
+      "/\\*\\s+@overrideSelector\\s+\\{(.*)\\}\\s+\\*/");
 
   private static final Pattern OVERRIDE_DEF_REGEX = Pattern.compile(
-      "/\\*\\*?\\s+@overrideDef\\s+\\{(.*)\\}\\s+\\*/");
+      "/\\*\\s+@overrideDef\\s+\\{(.*)\\}\\s+\\*/");
 
   private final VisitController visitController;
   private final ErrorManager errorManager;

--- a/src/com/google/common/css/compiler/passes/CollectProvideNamespaces.java
+++ b/src/com/google/common/css/compiler/passes/CollectProvideNamespaces.java
@@ -17,8 +17,6 @@
 package com.google.common.css.compiler.passes;
 
 import com.google.common.base.Preconditions;
-import com.google.common.collect.ImmutableListMultimap;
-import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.LinkedListMultimap;
 import com.google.common.collect.ListMultimap;
 import com.google.common.collect.Maps;
@@ -38,6 +36,8 @@ import java.util.Map;
  * This pass simply collects namespaces that correpond to constant definitions and mixins.
  * Also see the CheckMissingRequire pass that is used in conjunction with this one.
  *
+ * NOTE: The maps in this class can only be used within the same set of pass runs.
+ *
  */
 public final class CollectProvideNamespaces extends DefaultTreeVisitor implements CssCompilerPass {
   private final VisitController visitController;
@@ -54,19 +54,19 @@ public final class CollectProvideNamespaces extends DefaultTreeVisitor implement
   private final ListMultimap<String, String> defmixinProvideMap = LinkedListMultimap.create();
 
   public Map<String, String> getFilenameProvideMap() {
-    return ImmutableMap.copyOf(filenameProvideMap);
+    return filenameProvideMap;
   }
 
   public ListMultimap<String, String> getFilenameRequireMap() {
-    return ImmutableListMultimap.copyOf(filenameRequireMap);
+    return filenameRequireMap;
   }
 
   public ListMultimap<String, String> getDefProvideMap() {
-    return ImmutableListMultimap.copyOf(defProvideMap);
+    return defProvideMap;
   }
 
   public ListMultimap<String, String> getDefmixinProvideMap() {
-    return ImmutableListMultimap.copyOf(defmixinProvideMap);
+    return defmixinProvideMap;
   }
 
   public CollectProvideNamespaces(VisitController visitController) {
@@ -124,6 +124,10 @@ public final class CollectProvideNamespaces extends DefaultTreeVisitor implement
 
   @Override
   public void runPass() {
+    filenameProvideMap.clear();
+    filenameRequireMap.clear();
+    defProvideMap.clear();
+    defmixinProvideMap.clear();
     visitController.startVisit(this);
   }
 }

--- a/src/com/google/common/css/compiler/passes/CollectProvideNamespaces.java
+++ b/src/com/google/common/css/compiler/passes/CollectProvideNamespaces.java
@@ -28,12 +28,9 @@ import com.google.common.css.compiler.ast.CssMixinDefinitionNode;
 import com.google.common.css.compiler.ast.CssProvideNode;
 import com.google.common.css.compiler.ast.CssRequireNode;
 import com.google.common.css.compiler.ast.DefaultTreeVisitor;
-import com.google.common.css.compiler.ast.ErrorManager;
 import com.google.common.css.compiler.ast.VisitController;
 
 import java.util.Map;
-import java.util.logging.Logger;
-import java.util.regex.Pattern;
 
 /**
  * A compiler pass to help find missing {@code @require} lines for def constant references
@@ -43,13 +40,7 @@ import java.util.regex.Pattern;
  *
  */
 public final class CollectProvideNamespaces extends DefaultTreeVisitor implements CssCompilerPass {
-  private static final Logger logger = Logger.getLogger(CollectProvideNamespaces.class.getName());
-
-  private static final Pattern OVERRIDE_REGEX = Pattern.compile(
-      "/\\*\\s+@overrideSelector\\s+\\{(.*)\\}\\s+\\*/");
-
   private final VisitController visitController;
-  private final ErrorManager errorManager;
 
   // Key: filename; Value: provide namespace
   private final Map<String, String> filenameProvideMap = Maps.newHashMap();
@@ -78,10 +69,8 @@ public final class CollectProvideNamespaces extends DefaultTreeVisitor implement
     return ImmutableListMultimap.copyOf(defmixinProvideMap);
   }
 
-  public CollectProvideNamespaces(VisitController visitController,
-      ErrorManager errorManager) {
+  public CollectProvideNamespaces(VisitController visitController) {
     this.visitController = visitController;
-    this.errorManager = errorManager;
   }
 
   @Override

--- a/src/com/google/common/css/compiler/passes/DefaultGssSourceMapGenerator.java
+++ b/src/com/google/common/css/compiler/passes/DefaultGssSourceMapGenerator.java
@@ -32,7 +32,6 @@ import com.google.debugging.sourcemap.SourceMapGeneratorV3;
 import java.io.IOException;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.Deque;
 import java.util.List;
 import java.util.Map;
@@ -177,80 +176,54 @@ public final class DefaultGssSourceMapGenerator implements GssSourceMapGenerator
     ((SourceMapGeneratorV3) generator).setSourceRoot(path);
   }
 
+  private void addMapping(Mapping mapping) {
+    CssNode node = mapping.node;
+    FilePosition outputStartPosition = mapping.start;
+    FilePosition outputEndPosition = mapping.end;
+    String sourceFile = getSourceFileName(node);
+
+    // If the node does not have an associated source file or source location
+    // is unknown, then the node does not have sufficient info for source map.
+    if (sourceFile == null || node.getSourceCodeLocation().isUnknown()) {
+      return;
+    }
+
+    // TODO: could pass in an optional symbol name
+    generator.addMapping(
+        sourceFile, null,
+        new FilePosition(getStartLineno(node), getStartCharIndex(node)),
+        outputStartPosition, outputEndPosition);
+  }
+
+  /**
+   * Gets the source file file for current node.
+   */
+  private String getSourceFileName(CssNode node) {
+    return node.getSourceCodeLocation().getSourceCode().getFileName();
+  }
+
+  /**
+   * Gets the start line index in the source code of {@code node} adjusted to 0-based indices.
+   *
+   * <p>Notes: Gss compiler uses a 1-based line number and source map V3 uses a 0-based line number.
+   */
+  private int getStartLineno(CssNode node) {
+    return node.getSourceCodeLocation().getLineNumber() - 1;
+  }
+
+  /**
+   * Gets the start character index in the output buffer for current {@code node}.
+   */
+  private int getStartCharIndex(CssNode node) {
+    return node.getSourceCodeLocation().getCharacterIndex();
+  }
+
   /**
    * Generates the source map by passing all mappings to {@link #generator}.
    */
-  private void generateSourceMap() {
-    List<CompleteMapping> completeMappings = new ArrayList<>(allMappings.size());
+  private void generateSourceMap(){
     for (Mapping mapping : allMappings) {
-      // If the node does not have an associated source file or source location
-      // is unknown, then the node does not have sufficient info for source map.
-      if (mapping.node.getSourceCodeLocation().isUnknown()) {
-        continue;
-      }
-      CompleteMapping completeMapping = new CompleteMapping(mapping);
-      if (completeMapping.sourceFile == null) {
-        continue;
-      }
-      completeMappings.add(completeMapping);
-    }
-    Collections.sort(completeMappings);
-    for (CompleteMapping completeMapping : completeMappings) {
-      // TODO: could pass in an optional symbol name
-      generator.addMapping(
-          completeMapping.sourceFile, null,
-          completeMapping.inputStart,
-          completeMapping.outputStart, completeMapping.outputEnd);
-    }
-  }
-
-
-  private static final class CompleteMapping implements Comparable<CompleteMapping> {
-    final String sourceFile;
-    final FilePosition inputStart;
-    final FilePosition outputStart;
-    final FilePosition outputEnd;
-
-    CompleteMapping(Mapping mapping) {
-      CssNode node = mapping.node;
-      this.sourceFile = getSourceFileName(node);
-      this.inputStart = new FilePosition(
-          getStartLineno(node), getStartCharIndex(node));
-      this.outputStart = mapping.start;
-      this.outputEnd = mapping.end;
-    }
-
-    @Override
-    public int compareTo(CompleteMapping m) {
-      int delta = outputStart.getLine() - m.outputStart.getLine();
-      if (delta == 0) {
-        delta = outputStart.getColumn() - m.outputStart.getColumn();
-      }
-      return delta;
-    }
-
-    /**
-     * Gets the source file file for current node.
-     */
-    private static String getSourceFileName(CssNode node) {
-      return node.getSourceCodeLocation().getSourceCode().getFileName();
-    }
-
-    /**
-     * Gets the start line index in the source code of {@code node} adjusted to 0-based indices.
-     *
-     * <p>
-     * Note: Gss compiler uses a 1-based line number and source map V3 uses a 0-based line number.
-     */
-    private static int getStartLineno(CssNode node) {
-      return node.getSourceCodeLocation().getLineNumber() - 1;
-    }
-
-    /**
-     * Gets the start character index in the output buffer for current {@code node}.
-     */
-    private static int getStartCharIndex(CssNode node) {
-      return node.getSourceCodeLocation().getCharacterIndex();
+      addMapping(mapping);
     }
   }
 }

--- a/src/com/google/common/css/compiler/passes/DefaultGssSourceMapGenerator.java
+++ b/src/com/google/common/css/compiler/passes/DefaultGssSourceMapGenerator.java
@@ -32,6 +32,7 @@ import com.google.debugging.sourcemap.SourceMapGeneratorV3;
 import java.io.IOException;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Deque;
 import java.util.List;
 import java.util.Map;
@@ -176,54 +177,80 @@ public final class DefaultGssSourceMapGenerator implements GssSourceMapGenerator
     ((SourceMapGeneratorV3) generator).setSourceRoot(path);
   }
 
-  private void addMapping(Mapping mapping) {
-    CssNode node = mapping.node;
-    FilePosition outputStartPosition = mapping.start;
-    FilePosition outputEndPosition = mapping.end;
-    String sourceFile = getSourceFileName(node);
-
-    // If the node does not have an associated source file or source location
-    // is unknown, then the node does not have sufficient info for source map.
-    if (sourceFile == null || node.getSourceCodeLocation().isUnknown()) {
-      return;
-    }
-
-    // TODO: could pass in an optional symbol name
-    generator.addMapping(
-        sourceFile, null,
-        new FilePosition(getStartLineno(node), getStartCharIndex(node)),
-        outputStartPosition, outputEndPosition);
-  }
-
-  /**
-   * Gets the source file file for current node.
-   */
-  private String getSourceFileName(CssNode node) {
-    return node.getSourceCodeLocation().getSourceCode().getFileName();
-  }
-
-  /**
-   * Gets the start line index in the source code of {@code node} adjusted to 0-based indices.
-   *
-   * <p>Notes: Gss compiler uses a 1-based line number and source map V3 uses a 0-based line number.
-   */
-  private int getStartLineno(CssNode node) {
-    return node.getSourceCodeLocation().getLineNumber() - 1;
-  }
-
-  /**
-   * Gets the start character index in the output buffer for current {@code node}.
-   */
-  private int getStartCharIndex(CssNode node) {
-    return node.getSourceCodeLocation().getCharacterIndex();
-  }
-
   /**
    * Generates the source map by passing all mappings to {@link #generator}.
    */
-  private void generateSourceMap(){
+  private void generateSourceMap() {
+    List<CompleteMapping> completeMappings = new ArrayList<>(allMappings.size());
     for (Mapping mapping : allMappings) {
-      addMapping(mapping);
+      // If the node does not have an associated source file or source location
+      // is unknown, then the node does not have sufficient info for source map.
+      if (mapping.node.getSourceCodeLocation().isUnknown()) {
+        continue;
+      }
+      CompleteMapping completeMapping = new CompleteMapping(mapping);
+      if (completeMapping.sourceFile == null) {
+        continue;
+      }
+      completeMappings.add(completeMapping);
+    }
+    Collections.sort(completeMappings);
+    for (CompleteMapping completeMapping : completeMappings) {
+      // TODO: could pass in an optional symbol name
+      generator.addMapping(
+          completeMapping.sourceFile, null,
+          completeMapping.inputStart,
+          completeMapping.outputStart, completeMapping.outputEnd);
+    }
+  }
+
+
+  private static final class CompleteMapping implements Comparable<CompleteMapping> {
+    final String sourceFile;
+    final FilePosition inputStart;
+    final FilePosition outputStart;
+    final FilePosition outputEnd;
+
+    CompleteMapping(Mapping mapping) {
+      CssNode node = mapping.node;
+      this.sourceFile = getSourceFileName(node);
+      this.inputStart = new FilePosition(
+          getStartLineno(node), getStartCharIndex(node));
+      this.outputStart = mapping.start;
+      this.outputEnd = mapping.end;
+    }
+
+    @Override
+    public int compareTo(CompleteMapping m) {
+      int delta = outputStart.getLine() - m.outputStart.getLine();
+      if (delta == 0) {
+        delta = outputStart.getColumn() - m.outputStart.getColumn();
+      }
+      return delta;
+    }
+
+    /**
+     * Gets the source file file for current node.
+     */
+    private static String getSourceFileName(CssNode node) {
+      return node.getSourceCodeLocation().getSourceCode().getFileName();
+    }
+
+    /**
+     * Gets the start line index in the source code of {@code node} adjusted to 0-based indices.
+     *
+     * <p>
+     * Note: Gss compiler uses a 1-based line number and source map V3 uses a 0-based line number.
+     */
+    private static int getStartLineno(CssNode node) {
+      return node.getSourceCodeLocation().getLineNumber() - 1;
+    }
+
+    /**
+     * Gets the start character index in the output buffer for current {@code node}.
+     */
+    private static int getStartCharIndex(CssNode node) {
+      return node.getSourceCodeLocation().getCharacterIndex();
     }
   }
 }

--- a/src/com/google/common/css/compiler/passes/ProcessComponents.java
+++ b/src/com/google/common/css/compiler/passes/ProcessComponents.java
@@ -219,6 +219,7 @@ public class ProcessComponents<T> extends DefaultTreeVisitor
       Set<String> constants, CssComponentNode target, CssComponentNode source) {
     CssBlockNode sourceBlock = source.getBlock();
     CssBlockNode copyBlock = new CssBlockNode(false, sourceBlock.deepCopy().getChildren());
+    copyBlock.setSourceCodeLocation(source.getBlock().getSourceCodeLocation());
     CssTree tree = new CssTree(
         target.getSourceCodeLocation().getSourceCode(), new CssRootNode(copyBlock));
     new TransformNodes(constants, target, target != source,

--- a/src/com/google/common/css/compiler/passes/SplitRulesetNodes.java
+++ b/src/com/google/common/css/compiler/passes/SplitRulesetNodes.java
@@ -61,6 +61,7 @@ public class SplitRulesetNodes extends SkippingTreeVisitor
       for (CssSelectorNode sel : selectors.childIterable()) {
         for (CssNode child : declarations.childIterable()) {
           CssRulesetNode ruleset = new CssRulesetNode();
+          ruleset.setSourceCodeLocation(node.getSourceCodeLocation());
           ruleset.addDeclaration(child.deepCopy());
           ruleset.addSelector(sel.deepCopy());
 

--- a/tests/com/google/common/css/compiler/ast/testing/NewFunctionalTestBase.java
+++ b/tests/com/google/common/css/compiler/ast/testing/NewFunctionalTestBase.java
@@ -243,7 +243,9 @@ public class NewFunctionalTestBase extends FunctionalTestCommonBase {
       if (exactMatch) {
         assertEquals(expectedMessages[currentIndex], message);
       } else {
-        assertTrue(message.contains(expectedMessages[currentIndex]));
+        assertTrue("Expected error '" + message + "' to contain '"
+            + expectedMessages[currentIndex] + "'.",
+            message.contains(expectedMessages[currentIndex]));
       }
       currentIndex++;
     }

--- a/tests/com/google/common/css/compiler/passes/CheckMissingRequireTest.java
+++ b/tests/com/google/common/css/compiler/passes/CheckMissingRequireTest.java
@@ -42,7 +42,7 @@ public class CheckMissingRequireTest extends NewFunctionalTestBase {
       pass.runPass();
     }
     CollectProvideNamespaces collectProvides = new CollectProvideNamespaces(
-        tree.getVisitController(), errorMgr);
+        tree.getVisitController());
     collectProvides.runPass();
     new CheckMissingRequire(
         tree.getVisitController(),

